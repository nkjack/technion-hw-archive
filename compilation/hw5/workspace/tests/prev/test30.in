void Func(int i, int k, byte c) {
	printi(i);
	print("\n");
	printi(k);
	print("\n");
	printi(c);
}
void maxTestsPassMultipleParametersToFunc() {
	int t = 33;
	byte s = 3b;
	Func(8 + 2, t, s + 8b);
	print("\n");
	Func(14, t, s + 8b);
}


void main() {
     maxTestsPassMultipleParametersToFunc();
     print("\nI hate this course\n");
}
