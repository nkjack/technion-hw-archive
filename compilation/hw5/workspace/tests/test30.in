int doSomething(byte x, bool c) {
	if (c) {
		int i = 0;
		int switch = 1;
		while (i < x) {
			switch = switch * x;
			i = i + 1;
            
            if (switch < 1000 and i / 5 == 0) {
                printi(i * 256);
                continue;
                break;
            }
            
			if (switch > 1000)
				return switch;
            else
                continue;
		}
		return switch * 10;
	}
	else {
		return x - 50;
	}
}

void main() {
	print("10,true: ");
	printi(doSomething(10b,true));
	print("\n50,true: ");
	printi(doSomething(50b,true));
	print("\n10,false: ");
	printi(doSomething(10b,false));
	print("\n50,false: ");
	printi(doSomething(50b,false));
	print("\n");
}