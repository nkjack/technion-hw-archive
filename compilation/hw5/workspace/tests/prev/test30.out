Loaded: ./exceptions.s
10
33
11
14
33
11
I hate this course
